File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/core/handler/EnableNamespaceHandler.py
from javonet.core.handler.AbstractCommandHandler import AbstractCommandHandler
from javonet.core.namespaceCache.NamespaceCache import NamespaceCache


class EnableNamespaceHandler(AbstractCommandHandler):
    def __init__(self):
        self._required_parameters_count = 1

    def process(self, command):
        try:
            if len(command.payload) < self._required_parameters_count:
                raise Exception(self.__class__.__name__ + " parameters mismatch!")

            namespace_cache = NamespaceCache()

            for payload in command.payload:
                if isinstance(payload, str):
                    namespace_cache.cache_namespace(payload)
                if isinstance(payload, list):
                    for namespace_to_enable in payload:
                        namespace_cache.cache_namespace(namespace_to_enable)

            return 0

        except Exception as e:
            exc_type, exc_value = type(e), e
            new_exc = exc_type(exc_value).with_traceback(e.__traceback__)
            raise new_exc from None

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Nodejs/core/handler/EnableNamespaceHandler.js
const AbstractHandler = require("./AbstractHandler");
const NamespaceCache = require("../namespaceCache/NamespaceCache");

class EnableNamespaceHandler extends AbstractHandler {

    constructor() {
        super()
        this.requiredParametersCount = 1
    }

    process(command) {
        try {
            if (command.payload.length < this.requiredParametersCount) {
                throw new Error(this.constructor.name + " parameters mismatch")
            }
            const namespace_cache = new NamespaceCache();

            for (let payload of command.payload) {
                if (typeof payload === 'string') {
                    namespace_cache.cacheNamespace(payload);
                }
                if (Array.isArray(payload)) {
                    for (let namespace_to_enable of payload) {
                        namespace_cache.cacheNamespace(namespace_to_enable);
                    }
                }
            }
            return 0;

        } catch (error) {
            throw this.process_stack_trace(error, this.constructor.name)
        }

    }
}

module.exports = new EnableNamespaceHandler()